salt/modules/cohesity.py

```diff
@@ -0,0 +1,416 @@
+    resp = cohesity_client.view_boxes.get_view_boxes(names=name)
+    if resp:
+        return resp[0].id
+
+
+def get_policy_id(name):
+    """
+    Function to fetch policy available in the cluster.
+    : return policy id.
+    """
+    logger.info('Getting policy with name %s' % name)
+    resp = cohesity_client.protection_policies.get_protection_policies(
+        names=name)
+    if resp:
+        return resp[0].id
+
+
+def get_vmware_source_ids(name, vm_list):
+    """
+    Function to virtual machines available in the vcenter.
+    : return source ids and vcenter id.
+    """
+    source_id_list = []
+    parent_id = -1
+    logger.info('Fetching Vcenter and Vm ids')
+    try:
+        result = cohesity_client.protection_sources\
+            .list_protection_sources_root_nodes(environments=env_enum.K_VMWARE)
+        for each_source in result:
+            if each_source.registration_info.access_info.endpoint == name:
+                parent_id = each_source.protection_source.id
+        if parent_id == -1:
+            logger.error('Vcenter %s not available in the cluster' % name)
+            exit()
+        vms = cohesity_client.protection_sources.list_virtual_machines(
+            v_center_id=parent_id, names=vm_list)
+        vm_names = [vm_list] if type(vm_list) == str else copy.deepcopy(
+            vm_list)
+        for vm in vms:
+            vm_names.remove(vm.name)
+            source_id_list.append(vm.id)
+        if vm_names:
+            return 'Following list of vms %s are not available in vcenter, ' \
+            'please make sure the virtual machine names are correct' \
+            % ','.join(vm_names)
+        return parent_id, source_id_list
+    except APIException as err:
+        logger.error(err)
+        return -1, []
+    except Exception as err:
+        logger.error(err)
+        return -1, []
+
+
+def register_vcenter(vcenter, username, password):
+    """
+    Function to fetch register Vcenter.
+    """
+    try:
+        existing_sources = cohesity_client.protection_sources.\
+            list_protection_sources_root_nodes(environment=env_enum.K_VMWARE)
+        for source in existing_sources:
+            if source.registration_info.access_info.endpoint == vcenter:
+                return 'Source with name %s is already registered' % vcenter
+        body = RegisterProtectionSourceParameters()
+        body.endpoint = vcenter
+        body.environment = env_enum.K_VMWARE
+        body.username = username
+        body.password = password
+        body.vmware_type = 'kVCenter'
+        cohesity_client.protection_sources.create_register_protection_source(
+            body)
+        return 'Successfully registered Vcenter %s' % vcenter
+    except Exception as err:
+        logger.error(err)
+        return str(err)
+
+
+def create_vmware_protection_job(job_name, vcenter_name, sources,
+    policy='Gold', domain='DefaultStorageDomain', pause_job=True,
+    timezone='Europe/Berlin', description=''):
```

Comment:
Use local time from where salt is running if not specified :  
`'/'.join(os.path.realpath('/etc/localtime').split('/')[-2:]) `